st2actions/st2actions/workflows/workflows.py

```diff
@@ -44,6 +51,7 @@
 class WorkflowExecutionHandler(consumers.VariableMessageHandler):
     def __init__(self, connection, queues):
         super(WorkflowExecutionHandler, self).__init__(connection, queues)
+        self._running_workflows = list()
```

Comment:
This list is not sorted. I can see performance issue trying to search/delete from this list if there are a lot of workflows being processed. Can this be a dict or some other data structure w/ constant lookup time?  Also, I don't see any lock mechanism, this is not thread safe. Since the workflow engine at any given point will be processing multiple workflows and updating this list, it's going to cause issue w/o some sort of coordination.